herbert_core/classes/MPIFramework/@JobDispatcher/private/display_fail_jobs_.m

```diff
@@ -1,107 +1,143 @@
     end
+
 end
-if any(mEXceptions_outputs)
-    if isempty(Err_code)
-        warn_code = 'DISPLAY_FAIL_JOBS:parallel_failure';
-    else
-        warn_code = Err_code;
-    end
-    warning(warn_code ,...
-        ' Number: %d parallel tasks out of total: %d tasks have failed',...
-        n_failed,n_workers)
-    errOutputs = outputs(mEXceptions_outputs);
-    if iscell(errOutputs)
-        for i=1:numel(errOutputs)
-            disp(['***** Error output N ',num2str(i)]);
-            disp_exception(errOutputs{i});
+
+if any(MExceptions_outputs)
+
+    warning(warn_code, '%d of %d tasks have failed', n_failed, n_workers)
+
+    err_outputs = outputs(MExceptions_outputs);
+
+    if iscell(err_outputs)
+        for i=1:numel(err_outputs)
+            debug_print(['***** Error output N ', num2str(i)]);
+            disp_exception(err_outputs{i});
         end
     else
-        disp_exception(errOutputs);
+        disp_exception(err_outputs);
     end
-    if ~isempty(Err_code)
-        error(Err_code,'Parallel job have failed, producing errors above.');
-    end
-else
-    if ~isempty(Err_code)
-        error(Err_code,...
-            ' Number: %d parallel tasks out of total: %d tasks have failed without returning the reason',...
-            n_failed,n_workers)
+    if ~isempty(err_code)
+        error(err_code, 'Parallel job failed, producing the errors above.');
     end
+elseif ~isempty(err_code)
+        error(err_code, ...
+              '%d of %d tasks have failed without returning the reason', ...
+              n_failed, n_workers)
 end
 
-function disp_exception(errOutput)
-%
-if isa(errOutput,'MException')
-    disp(getReport(errOutput))
-elseif iscell(errOutput)
-    disp('***************************************************************');
-    disp(errOutput);
-    for i=1:numel(errOutput)
-        sprintf(' Cell %d, contains: %s\n',i,evalc('disp(errOutput{i}'));
-        disp_exception(errOutput{i});
+if ~debug
+    fclose(fh);
+end
+
+
+end
+
+
+function disp_exception(err_output)
+
+if isa(err_output, 'MException')
+    debug_print(disp2str(getReport(err_output)))
+
+elseif iscell(err_output)
+
+    for i=1:numel(err_output)
+        debug_print(' Cell %d, contains: \n', i);
+        disp_exception(err_output{i});
     end
-    disp('***************************************************************');
-elseif isfield(errOutput,'error') && isa(errOutput.error,'MException')
-    for i=1:numel(errOutput.error)
-        disp(getReport(errOutput.error(i)));
+
+elseif isfield(err_output, 'error') && isa(err_output.error, 'MException')
+
+    for i=1:numel(err_output.error)
+        debug_print(disp2str(getReport(err_output.error(i))));
     end
+
 else
-    disp('unknown type of error returned')
+    debug_print('unknown type of error: \n %s', disp2str(err_output));
+
 end
 
+end
 
+function debug_print(varargin)
+    if debug
+        fprintf(varargin{:});
+    else
+        fprintf(fh, varargin{:});
+    end
+end
```

Comment:
completely agree with need for test. Absolutely not-obvious that `fh` is global variable and it propagated correctly.

The idea looks reasonable.